test/test_overrides.py

```diff
@@ -804,7 +804,6 @@ def test_gradcheck(self):
         # Tensor-likes.
         self.assertEqual(total_used_attrs, {
             'data',
-            'device',
```

Comment:
You might be surprised that a refactor will involve modifying a test. That is because the previous behavior was a bug. Previously, gradcheck may actually return True when `raise_exception` is False and we fail `torch.allclose`. However, when we fix that behavior in this refactor by returning the output of `checkIfNumericalAnalyticAreClose` (see below), we short-circuit our function such that `.device` is no longer an accessed attr.